### ISSUE TITLE
`Lint/Void` and unary operators

### ISSUE THREAD

**Issue by kivikakk** [rubocop/rubocop#13818]
We're having a bit of a discussion over at https://github.com/minimagick/minimagick/issues/584 as to `Lint/Void` and an effectful `+` or `+@` definition.

The issue at hand is that it's considered a "safe" cop which is corrected by `--safe`. In this case, `+` and `+@` are giving commands directly to an invocation of `ImageMagick`; see [an example in the README](https://github.com/minimagick/minimagick?tab=readme-ov-file#plus-options):

> ### "Plus" options
> 
> ```rb
> MiniMagick.convert do |convert|
>   convert << "input.jpg"
>   convert.repage.+
>   convert.distort.+("Perspective", "more args")
> end
> ```
> ```
> magick input.jpg +repage +distort Perspective 'more args'
> ```

Running Rubocop with `--safe` will change these invocations to `convert.repage` and `convert.distort`, which in turn modifies the resulting command.

There isn't a very clear way to avoid this, as detailed by a maintainer over in that issue, and while it may be unusual, I think it's valid to question the assumption that all prefix/infix operators are effect-free and safe to change automatically.

Thoughts appreciated!

**Comment by vlad-pisanov** [comment]
I think a lot of "safe" cops are actually unsafe if we consider the possibility of redefining `+`, `==` and such. 😬 Like [Lint/NumericOperationWithConstantResult](https://docs.rubocop.org/rubocop/cops_lint.html#lintnumericoperationwithconstantresult) or [Lint/UselessNumericOperation](https://docs.rubocop.org/rubocop/cops_lint.html#lintuselessnumericoperation)
